test/cxx/Core/ApplicationPool/PoolTest.cpp

```diff
@@ -2026,4 +2026,29 @@ namespace tut {
 
 
 	/*****************************/
+
+	TEST_METHOD(86) {
+		// If a request is in the getWaitlist for longer than maxRequestQueueTime,
+		// then an exception is returned.
+		Options options = createOptions();
+		options.appGroupName = "test1";
+                options.maxRequestQueueSize = 2;
+		options.maxRequestQueueTime = 1;
```

Comment:
I think it is a good idea to make Options.maxRequestQueueTime milliseconds resolution. That way you can limit the time spent waiting in unit tests to a few tens or few hundred milliseconds. Users should be exposed to seconds resolution though, so don't forget to make the Nginx/Apache/Standalone configuration have seconds resolution. See `passenger_start_timeout` for an example.